filters/extract_area.go

```diff
@@ -0,0 +1,121 @@
+package filters
+
+import (
+	"errors"
+	"fmt"
+	"github.com/zalando/skipper/filters"
+	"gopkg.in/h2non/bimg.v1"
+	"strconv"
+	"strings"
+)
+
+const (
+	ExtractArea = "extractArea"
+	cropFrom    = "point"
+	cropWidth   = "width"
+	cropHeight  = "height"
+)
+
+type extractArea struct{}
+
+func NewExtractArea() filters.Spec {
+	return &extractArea{}
+}
+
+func (t *extractArea) Name() string {
+	return ExtractArea
+}
+
+func (t *extractArea) CreateFilter(config []interface{}) (filters.Filter, error) {
+	return t, nil
+}
+
+func (t *extractArea) CreateOptions(img *bimg.Image, options map[string][]string) (*bimg.Options, error) {
+
+	// GET starting position to crop from
+	point, ok := options[cropFrom]
+	if !ok {
+		return nil, errors.New(fmt.Sprintf("could not find information about where to start the crop from."))
```

Comment:
The reason we error on not getting options, is because, until now, the options were configured (in the eskip file). The goal to return error was to make sure the options were configured correctly.

In this case, the options are input by the user. Because of that, I don't think returning error when a parameter is missing is a good practice.

What I would rather do, is create a Skrop filter called something like `queryParamTransformation`. In the CreateOptions method, based on the query parameters, I would try to figure out what kind of transformation our user wants to do. For example if the user provides 3 parameters: `cropFrom`, `width`, `height` I would know that she wants to crop the image.

Reply:
We could meet to discuss about this.

Reply:
sure. I will come by to your place or ping you tomorrow.